mne/epochs.py

```diff
@@ -818,6 +869,59 @@ def _compute_mean_or_stderr(self, picks, mode='ave'):
         return self._evoked_from_epoch_data(data, self.info, picks, n_events,
                                             kind, self._name)
 
+    def regress(self, on, fit_intercept=True, by=None):
```

Comment:
IMO the linear_regression code is a bit arcane. I don't mind seeing an extra method. Although I don't like "on".
How close can we get to sklearn style? How about fit_transform(X)?